core/src/main/scala/org/apache/spark/rpc/netty/Outbox.scala

```diff
@@ -215,7 +216,7 @@ private[netty] class Outbox(nettyEnv: NettyRpcEnv, val address: RpcAddress) {
     // update messages and it's safe to just drain the queue.
     var message = messages.poll()
```

Comment:
But why? What's being optimized here? It's such a rare occurrence that it's better to pay the allocation cost when it happens than waste the memory to keep the singleton around.